src/main/java/com/sergiofah/model/Product.java

```diff
@@ -32,4 +34,18 @@ public class Product {
 
     @Column(name = "image_url")
     private String imageUrl;
+
+    @Override
+    public boolean equals(Object object) {
+        if (this == object) return true;
+        if (object == null) return false;
+
+        if(object.getClass() == ProductDTO.class){
+            ProductDTO productDTO = (ProductDTO) object;
+            if(this.id.equals(productDTO.getId())&&(this.category.getId().equals(productDTO.getCategoryId()))&&(this.line.getId().equals(productDTO.getLineId()))&&(this.model.equals(productDTO.getModel()))&&(this.description.equals(productDTO.getDescription())&&(this.imageUrl.equals(productDTO.getImageUrl())))){
+                return true;
+            }
+        }
```

Comment:
encontrei três pequenos pontos para correção e melhoria também:

1. Favor, não usar o operador "==" quando for comparar dois objetos, apenas quando for números. Isso porque o operador "==" mesmo sendo mais rápido (porém não é notório) ele não tem uma assertividade tão boa quanto usando o método equals(). Isso pode até ser explicado para o caso de sobrescrever o método equals(), imagine caso o método o assertArrayEquals usasse o operador "==" ao invés do método equals dos objetos que ele recebe como parâmetro.
2. Quanto ao retorno, sugiro realizar uma quebra de linha para melhorar a visualização do código e deve ser feita da maneira correta, para isso, sugiro dar uma lida sobre as convenções do Java, ele é importante pois cada linguagem possui uma maneira correta na hora de quebrar linhas, como escrever uma variavel e assim por diante. (https://www.oracle.com/java/technologies/javase/codeconventions-introduction.html)
3. Recomendo seguir o mesmo formato que fez no Line e Product de realizar o retorno logo com as condicionais dentro do if